src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/mapper/CaseDetailsMapper.java

```diff
@@ -3,12 +3,19 @@
 import java.util.List;
 
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
 import org.mapstruct.ReportingPolicy;
 import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
 import uk.gov.hmcts.ccd.domain.service.search.elasticsearch.dto.ElasticSearchCaseDetailsDTO;
 
 @Mapper(unmappedTargetPolicy = ReportingPolicy.IGNORE, componentModel = "spring")
 public interface CaseDetailsMapper {
 
+    String EMPTY_MAP = "java(new HashMap<>())";
+
+    @Mapping(source = "data", target = "data", defaultExpression = EMPTY_MAP)
+    @Mapping(source = "dataClassification", target = "dataClassification", defaultExpression = EMPTY_MAP)
```

Comment:
This will prevent NPEs in future with Global Search scenarios where a case doesn't have any of the "extra" fields in addition to `SearchCriteria`, as currently this will map to a `null` `data`.

Reply:
👍